src/processor/reconnection-graph.service.ts

```diff
@@ -49,23 +56,31 @@ export class ReconnectionGraphService implements OnApplicationBootstrap, OnAppli
 
     // TODO set state based on the response from getUserGraphFromProvider
     const [graphConnections, graphKeyPair] = await this.getUserGraphFromProvider(dsnpUserId, providerId);
-    this.logger.log('graphConnections', graphConnections);
-    this.logger.log('graphKeyPair', graphKeyPair);
 
     // graph config and respective schema ids
-    const graphSdkConfig = await this.graphStateManager.getGraphConfig();
-    const public_follow_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Follow, PrivacyType.Public);
-    const public_friendship_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Friendship, PrivacyType.Public);
-    const private_follow_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Follow, PrivacyType.Private);
-    const private_friendship_schema_id = await this.graphStateManager.getSchemaIdFromConfig(ConnectionType.Friendship, PrivacyType.Private);
+    const graphSdkConfig  = await this.graphStateManager.getGraphConfig();
+    
+    // get the user's DSNP Graph from the blockchain and form import bundles
+    const importBundles = await this.formImportBundles(dsnpUserId, graphSdkConfig, graphKeyPair);
+    await this.graphStateManager.importUserData(importBundles).then((results) => {
+      throw new Error(JSON.stringify("importUserData results: " + results));
+    });
+
+    let exportedUpdates: Update[] = [];
 
+    if (updateConnections) {
+      // using graphConnections form Action[] and update the user's DSNP Graph
+      const actions: ConnectAction[] = await this.formConnections(dsnpUserId, providerId, graphSdkConfig, graphConnections);
+      await this.graphStateManager.applyActions(actions).then((results) => {
+        throw new Error(JSON.stringify("applyActions results: " + results));
+      });
```

Comment:
See previous comment. Also, throwing an error from a `Promise.then(() => )` method will cause the main promise to reject, which in this example will cause the enclosing function to throw an error (thus next line will never be reached)